### PR TITLE
Add file controller authorization

### PR DESCRIPTION
**Pull request by morganbr**
Add authorization for navigating to files. Also fixes lazy loading proxies and file display to players

### CODE REVIEW

File: ServerCore/Pages/FilesController.cs [hunk 1]
```diff
@@ -35,8 +44,83 @@ public IActionResult Index(int eventId, string filename)
...
+            PuzzleUser user = PuzzleUser.GetPuzzleUserForCurrentUser(context, User, userManager);
+
+            // Admins can see all files
+            if (user.IsAdminForEvent(context, currentEvent))
+            {
+                return true;
+            }
+
+            // Authors can see all files attached to their puzzles
+            if (await UserEventHelper.IsAuthorOfPuzzle(context, puzzle, user))
```
Comment (asyasky) [inline ServerCore/Pages/FilesController.cs]: Can you just return this directly?

File: ServerCore/Pages/FilesController.cs [hunk 2]
```diff
@@ -35,8 +44,83 @@ public IActionResult Index(int eventId, string filename)
...
+            }
+
+            Team team = await UserEventHelper.GetTeamForPlayer(context, currentEvent, user);
+            if (team == null)
+            {
+                return false;
+            }
+
+            // Once answers are available, so are all other files
+            if (currentEvent.AreAnswersAvailableNow)
```
Comment (asyasky) [inline ServerCore/Pages/FilesController.cs]: Ditto

File: ServerCore/Pages/FilesController.cs (hunk 1 above)
Comment (morganbr) [reply ServerCore/Pages/FilesController.cs]: Nope, as discussed

File: ServerCore/Pages/FilesController.cs (hunk 2 above)
Comment (morganbr) [reply ServerCore/Pages/FilesController.cs]: Ditto

**Pull request merged**